### PR TITLE
React Components: Emphasise component capitalisation requirement

### PR DESCRIPTION
**Pull request by MaoShizhong**
## Because
React components **must** be capitalised to actually work as components. The current wording I felt was quite easy to miss/gloss over so it has been reworded to bring more attention to its necessity.

## This PR
- Reworded/added some short sentences to emphasise this point to the reader
- Fixed inconsistent list styling to follow the style guide

## Additional Information
N/A

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [ ] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: react/getting_started_with_react/react_components.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ function Greeting() {
 
 This might look mostly familiar to you - it's a simple JavaScript function, which returns JSX. Open up the project you were working on, create a new file named `Greeting.jsx`, and in that file write your own handmade functional component. Name it whatever you wish, and have it return whatever JSX you wish. 
 
-Are you done? Check the naming of your function! Is it capitalized? Keep this key difference in mind, components that aren't capitalized might not run as expected - which is why we capitalized `Greeting()`.
+Are you done? Check the naming of your function! Is it capitalized? Keep this key difference in mind. **React components must be capitalized** or they may not function as expected, which is why we capitalized `Greeting()`.
```
Comment (ManonLef) [inline react/getting_started_with_react/react_components.md:40]:
```suggestion
Are you done? Check the naming of your function! Is it capitalized? Keep this key difference in mind. **React components must be capitalized** or they may not function as expected, which is why we capitalized `Greeting()`.
```

Should we perhaps add why this happens somewhere in the lesson? A note about JSX parsing and its recognition as React components vs. regular HTML components/elements?

File: react/getting_started_with_react/react_components.md [hunk 2]
```diff
@@ -79,26 +79,26 @@ ReactDOM.createRoot(document.getElementById('root')).render(
 )
 ~~~
 
-And just like that, you've successfully imported and used your first custom-made component, congratulations!
+Remember that `<Greeting />` should be capitalized! Try using lower case for the import, function name and component and see what happens. Otherwise, just like that, you've successfully imported and used your first custom-made component, congratulations!
```
Comment (ManonLef) [inline react/getting_started_with_react/react_components.md:82]: perhaps it should be added here then. At this point JSX is explained :)

Comment (MaoShizhong) [reply react/getting_started_with_react/react_components.md:82]: Ah, that's actually a good point - flows even better!

**Review by ManonLef — APPROVED** [review]

**Comment by ManonLef** [comment]
Sorry, it's been a while and this apparently slipped my attention until I got assigned. 
Thanks so much for doing this!

**Pull request merged**